fdbserver/ConfigNode.actor.cpp

```diff
@@ -570,6 +575,42 @@ class ConfigNodeImpl {
 		}
 	}
 
+	ACTOR static Future<Void> serve(ConfigNodeImpl* self, ConfigBroadcastInterface const* cbi, bool infinite) {
+		loop {
+			choose {
+				when(state ConfigBroadcastRegisteredRequest req = waitNext(cbi->registered.getFuture())) {
+					bool isRegistered = wait(registered(self));
+					req.reply.send(ConfigBroadcastRegisteredReply{ isRegistered });
+				}
+				when(ConfigBroadcastReadyRequest readyReq = waitNext(cbi->ready.getFuture())) {
+					readyReq.reply.send(ConfigBroadcastReadyReply{});
+					if (!infinite) {
+						return Void();
+					}
+				}
+			}
+		}
+	}
+
+	ACTOR static Future<Void> serve(ConfigNodeImpl* self,
+	                                ConfigBroadcastInterface const* cbi,
+	                                ConfigTransactionInterface const* cti,
+	                                ConfigFollowerInterface const* cfi) {
+		wait(serve(self, cbi, false));
+
+		self->kvStore->set(KeyValueRef(registeredKey, BinaryWriter::toValue(true, IncludeVersion())));
+		wait(self->kvStore->commit());
+
+		// Shouldn't return (coordinationServer will throw an error if it does).
+		wait(serve(self, cbi, true) || serve(self, cti) || serve(self, cfi));
```

Comment:
Do the other 2 `ConfigNode::serve` overloads still need to be public methods if they're only used internally here?